lib/publiccloud/ec2.pm

```diff
@@ -130,6 +130,12 @@ sub upload_img {
     my $vpc_subnet = get_var('PUBLIC_CLOUD_EC2_UPLOAD_VPCSUBNET');
     my $instance_type = get_required_var('PUBLIC_CLOUD_EC2_UPLOAD_INSTANCE_TYPE');
 
+    if (!$vpc_subnet) {
+        my $vpc_id = script_output("aws ec2 describe-vpcs --region " . $self->provider_client->region . " --filters 'Name=tag:Name,Values=tf-vpc' --output text --query 'Vpcs[0].VpcId'");
+        # Note: We're grabbing first subnet, say eu-central-1a in eu-central-1
+        $vpc_subnet = script_output("aws ec2 describe-subnets --region " . $self->provider_client->region . " --filters 'Name=vpc-id,Values=$vpc_id' --output text --query 'Subnets[0].SubnetId'");
+    }
+
     # ec2uploadimg will fail without this file, but we can have it empty
     # because we passing all needed info via params anyway
     assert_script_run('echo " " > /root/.ec2utils.conf');
```

Comment:
For comment only, out scope: here wasn't used a simpler `touch file`: may be 'size' > 0 needed ?